### PR TITLE
Added option to automatically load game after start

### PR DESCRIPTION
**Pull request by Myszax**
Expansion for #33 . New option to load last quicksave immediately after game starts.

### CODE REVIEW

**Comment by Franisz** [comment]
Feels unnecessary and too problematic.

File: zUtilities/QuickSave.cpp [hunk 1]
```diff
@@ -279,12 +279,12 @@ namespace GOTHIC_ENGINE {
...
 
   void QuickSave::MenuLoop() {
-    if ( !Options::QuickSaveMode || zCMenu::inGameMenu || isLoading || !zinput->KeyToggled( Options::KeyQuickLoad ) ) return;
+    if ( !Options::QuickSaveMode || zCMenu::inGameMenu || isLoading ) return;
 
     oCSavegameInfo* info = gameMan->savegameManager->GetSavegame( iLastSaveSlot );
     if ( !info || !info->DoesSavegameExist() ) return;
 
-    LoadFromMainMenu();
+    if ( Options::LoadLastQuickSaveAfterGameStart || zinput->KeyToggled( Options::KeyQuickLoad ) ) LoadFromMainMenu();
```
Comment (folmert) [inline zUtilities/QuickSave.cpp:287]: @Myszax It's probably a cleaner solution than mine, but wouldn't it also trigger when user opens menu during game by pressing Esc? How is it prevented? `zCMenu::inGameMenu` checks if user is in main menu, or `MenuLoop` runs only in main menu?

Comment (the-overdriven) [inline zUtilities/QuickSave.cpp:287]: @Myszax It's probably a cleaner solution than mine (hooking to Game_Init with a delay), but wouldn't it also trigger when user opens menu during game by pressing Esc? How is it prevented? `zCMenu::inGameMenu` checks if user is in main menu, or `MenuLoop` runs only in main menu?

Comment (Myszax) [reply zUtilities/QuickSave.cpp:287]: `zCMenu::inGameMenu`:
is `true` when save has been loaded or new game has been started;
is `false` when gothic is running and no above action was executed

`MenuLoop` is running in main menu (before load/new game) and also when `ESC` is pressed when game is running (after load/new game).

In concluding this loop won't execute if save has been loaded/new game has been started.
Please read [#34](https://github.com/Franisz/zUtilities/pull/33#issuecomment-2304776242)
I hope it explains xD

Comment (the-overdriven) [reply zUtilities/QuickSave.cpp:287]: Then it seems fine, I don't see why it could be "problematic"

Comment (Franisz) [reply zUtilities/QuickSave.cpp:287]: In Gothic you cannot exit back to the menu once a save is loaded. How is that not problematic?

Comment (the-overdriven) [reply zUtilities/QuickSave.cpp:287]: I still don't understand the problem. Why would you need to go back to main menu? Autoload feature needs the user to be in main menu only once: after game is launched, but before save is loaded.

Comment (the-overdriven) [reply zUtilities/QuickSave.cpp:287]: Can you do something from main menu, that you can't from in-game menu? In-game menu is 1:1 the same as main menu. Even if it wasn't, there is always an option to disable the autoload feature.

Comment (Myszax) [reply zUtilities/QuickSave.cpp:287]: > In Gothic you cannot exit back to the menu once a save is loaded. How is that not problematic?

Why would you? :P When you enable in `gothic.ini` option to auto load after game start (which is disabled by default [in this pr]) what is the point to go back to main menu? You still have menu by pressing `ESC` in game when save is already loaded.

Comment (Franisz) [inline zUtilities/QuickSave.cpp:287]: Move condition before save check.

File: zUtilities/QuickSave.h [hunk 2]
```diff
@@ -3,11 +3,12 @@
 
 namespace GOTHIC_ENGINE {
   namespace Options {
-    int QuickSaveMode, UseBinarySave, KeyQuickSave, KeyQuickLoad, MinSaveSlot, MaxSaveSlot;
+    int QuickSaveMode, LoadLastQuickSaveAfterGameStart, UseBinarySave, KeyQuickSave, KeyQuickLoad, MinSaveSlot, MaxSaveSlot;
```
Comment (Franisz) [inline zUtilities/QuickSave.h:6]: Rename option to `QuickLoadAfterGameStart` and update project readme.

**Review by Franisz — CHANGES_REQUESTED** [review]

File: zUtilities/QuickSave.cpp (hunk 1 above)
Comment (Myszax) [reply zUtilities/QuickSave.cpp:287]: You mean `if ( !info || !info->DoesSavegameExist() ) return;` ?
I'm curious why? When save does not exists it's better to return than try to load even if `LoadLastQuickSaveAfterGameStart ` set to `true`.

Comment (Franisz) [reply zUtilities/QuickSave.cpp:287]: There is no need to check for the save existence every frame. First, we should check if the game should attempt to load at all.

Comment (Myszax) [reply zUtilities/QuickSave.cpp]: OK, you seems right.

File: README.md [hunk 3]
```diff
@@ -15,7 +15,8 @@ This is a plugin with a set of many quality-of-life and utility features made fo
   - Notice strings are set automatically depending on system language but still can be changed manually in `gothic.ini` with `CantSave`, `CantLoad` and `NoSave` options.
   - This feature can be disabled in `gothic.ini` with `QuickSaveMode` option.
   - Change `QuickSaveMode` option for different style or disable: `0` - _Disabled_, `1` - _Standard_, `2` - _Alternative_.
-  - Game can also be quick loaded from main menu.
+  - Game can also be quick loaded from main menu by pressing `KeyQuickLoad`.
+  - Save can bey automatically loaded after start when `QuickLoadAfterGameStart` is set to `1`.
```
Comment (Franisz) [inline README.md:19]: Fix typo and update also the section with default ini config.

Comment (Myszax) [reply README.md]: Done.

**Review by Franisz — APPROVED** [review]

**Pull request merged**